usaspending_api/database_scripts/job_archive/fix_awards_earliest_latest_transactions_and_exec_comp.sql

```diff
@@ -0,0 +1,150 @@
+        coalesce(fabs.officer_4_name, fpds.officer_4_name) as officer_4_name,
+        coalesce(fabs.officer_5_amount, fpds.officer_5_amount) as officer_5_amount,
+        coalesce(fabs.officer_5_name, fpds.officer_5_name) as officer_5_name
+    from
+        transaction_normalized as tn
+        left outer join transaction_fabs as fabs on fabs.transaction_id = tn.id
+        left outer join transaction_fpds as fpds on fpds.transaction_id = tn.id
+    where
+        coalesce(fabs.officer_1_name, fpds.officer_1_name) is not null
+    order by
+        tn.award_id,
+        tn.action_date desc,
+        tn.modification_number desc,
+        tn.transaction_unique_id desc
+),
+
+requiring_update as (
+    select
+        a.id as award_id,
+        e.transaction_id as earliest_transaction_id,
+        l.transaction_id as latest_transaction_id,
+        ec.officer_1_amount,
+        ec.officer_1_name,
+        ec.officer_2_amount,
+        ec.officer_2_name,
+        ec.officer_3_amount,
+        ec.officer_3_name,
+        ec.officer_4_amount,
+        ec.officer_4_name,
+        ec.officer_5_amount,
+        ec.officer_5_name
+    from
+        awards as a
+        left outer join earliest_transaction as e on e.award_id = a.id
+        left outer join latest_transaction as l on l.award_id = a.id
+        left outer join executive_compensation as ec on ec.award_id = a.id
+    where
+        a.earliest_transaction_id is distinct from e.transaction_id or
+        a.latest_transaction_id is distinct from l.transaction_id or
+        a.officer_1_amount is distinct from ec.officer_1_amount or
+        a.officer_1_name is distinct from ec.officer_1_name or
+        a.officer_2_amount is distinct from ec.officer_2_amount or
+        a.officer_2_name is distinct from ec.officer_2_name or
+        a.officer_3_amount is distinct from ec.officer_3_amount or
+        a.officer_3_name is distinct from ec.officer_3_name or
+        a.officer_4_amount is distinct from ec.officer_4_amount or
+        a.officer_4_name is distinct from ec.officer_4_name or
+        a.officer_5_amount is distinct from ec.officer_5_amount or
+        a.officer_5_name is distinct from ec.officer_5_name
+)
+
+update
+    awards as a
+set
+    earliest_transaction_id = e.id,
+    date_signed = e.action_date,
+    description = e.description,
+    period_of_performance_start_date = e.period_of_performance_start_date,
+    latest_transaction_id = l.id,
+    awarding_agency_id = l.awarding_agency_id,
+    category = case
+        when l.type in ('A', 'B', 'C', 'D') then 'contract'
+        when l.type in ('02', '03', '04', '05') then 'grant'
+        when l.type in ('06', '10') then 'direct payment'
+        when l.type in ('07', '08') then 'loans'
+        when l.type = '09' then 'insurance'
+        when l.type = '11' then 'other'
+        when l.type like 'IDV%' then 'idv'
+        else null
+    end,
+    certified_date = l.action_date,
```

Comment:
are we sure this is the correct mapping?

Reply:
`certified_date = l.action_date`? That's what the field currently is in award_helpers.py

Reply:
My apologies for not answering this sooner.  I was trying to find a kind way of asking "Why are you asking me this question when you could confirm it yourself in a matter of seconds?   Clearly I thought it was the correct mapping."

I could not come up with a kinder phrasing.

Reply:
I was just asking if it was a mistake on your part. Just to point it out so it would get a second look, since it wasn't an exact text match.